tests/series_only/__getitem___test.py

```diff
@@ -0,0 +1,18 @@
+from __future__ import annotations
+
+import polars as pl
+import pytest
+
+import narwhals.stable.v1 as nw
+
+
+def test_getitem() -> None:
+    spl = pl.Series([1, 2, 3])
+    assert spl[spl[0, 1]].equals(pl.Series([2, 3]))
```

Comment:
can we check what happens when there's missing values too please?

Reply:
Using `to_native()` instead of `to_numpy()` allows the following tests to pass, but my complete lack of knowledge on numpy, narwhals, polars and related subjects makes me unsure they are correct and show expected behavior:
```python
    spl = pl.Series([1, None, 3])
    assert spl[spl[0, 1]].equals(pl.Series([None, None]))

    snw = nw.from_native(spl, series_only=True)
    assert snw[snw[0, 1]].to_native().equals(pl.Series([None, None]))
```

What's your take on this?